packages/atomic/cypress/scripts/getHostedSearchPage.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Adds a sample hosted search page to the public folder for Cypress tests.
+ * Used for validating that the lastest Atomic build works with the hosted
+ * search pages.
+ */
+
+const fs = require('fs');
+const https = require('https');
+const jsdom = require('jsdom');
+const {JSDOM} = jsdom;
+const path = require('path');
+
+const SEARCH_URL = 'https://search.cloud.coveo.com';
+const ORG_ID = 'coveointernaltesting1';
+const PAGE_ID = '24b0c504-00d5-4afd-90cb-1b3e7e44040f';
+const PUBLIC_FOLDER = 'www/';
+const BASE_URL = `${SEARCH_URL}/rest/organizations/${ORG_ID}/searchinterfaces/${PAGE_ID}/`;
+
+/**
+ * Modifies the page to use the local version of Atomic
+ */
+function replaceAtomicVersion(document) {
+  const cssLink = document.querySelector('link[href$="themes/coveo.css"]');
+  const jsScript = document.querySelector('script[src$="atomic.esm.js"]');
+  cssLink.setAttribute('href', '/themes/coveo.css');
+  jsScript.setAttribute('src', '/build/atomic.esm.js');
+}
+
+/**
+ * Replaces the page init script to skip authentication
+ */
+function replaceInitScript(document) {
+  const initScript = document.querySelector('#bootstrap');
+  initScript.textContent = `
+    (async () => {
+      await customElements.whenDefined('atomic-search-interface')
+      searchInterface = document.querySelector('atomic-search-interface')
+      await searchInterface.initialize({
+        accessToken: 'xx564559b1-0045-48e1-953c-3addd1ee4457',
+        organizationId: 'searchuisamples'
+      });
+      searchInterface.executeFirstSearch()
+    })()
+  `;
+}
+
+/**
+ * Copies the css and js dependencies to the public folder
+ */
+function addCssAndJs() {
+  const links = [
+    'lib/styleguide/css/CoveoStyleGuide.css',
+    'css/main.css',
+    'js/SearchInterfaces.Dependencies.min.js',
+    'js/SearchInterfaces.bundle.min.js',
+  ];
+
+  links.forEach((link) => {
+    const filepath = PUBLIC_FOLDER + link;
+    try {
+      fs.mkdirSync(path.dirname(filepath), {recursive: true});
+    } catch (err) {
+      console.error(err);
+    }
```

Comment:
I'm not very good with error handling in js, please comment if it can be done better, here and elsewhere. I tried to use the synchronous methods where possible to make things simpler.

Reply:
It looks good, one question though, when an error occurs (the dir is not created), should we exit the process? or continue with the file downloads below?

Same comment for other catch statements (e.g. line 86)

Reply:
If any of the steps fail, the script might as well exit with failure immediately

Reply:
I think doing `process.exit` or throwing a new error are both ideal.

Reply:
ok, done ✔️